gelf/gelf.go

```diff
@@ -0,0 +1,74 @@
+package gelf
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aphistic/golf"
+	"github.com/yext/glog"
+	"github.com/yext/glog-contrib/raven"
+	"github.com/youtube/vitess/go/ratelimiter"
+)
+
+// Capture captures all events for a specific job and sends them to the gelf server.
+// Events sent at a higher rate than maxEventsPerSecond will be ignored.
+// The uri must have a udp or tcp scheme.
+func Capture(job, serverUri string, maxEventsPerSecond int, eventCh <-chan glog.Event) {
+	c, _ := golf.NewClient()
+	defer c.Close()
+
+	if c.Dial(serverUri) != nil {
+		panic("can't connect to gelf server")
+	}
+	logger, err := c.NewLogger()
+	if err != nil {
+		panic("can't initialize logger")
+	}
+
+	logger.SetAttr("job_name", job)
+
+	rl := ratelimiter.NewRateLimiter(maxEventsPerSecond, time.Second)
+	for e := range eventCh {
+		if !rl.Allow() {
+			continue
+		}
+
+		logEvent(logger, e)
+	}
+}
+
+func logEvent(logger *golf.Logger, e glog.Event) {
+	data := map[string]interface{}{}
+	for _, d := range e.Data {
+		switch t := d.(type) {
+		case map[string]interface{}:
+			for k, v := range t {
+				data[k] = v
+			}
+		}
+	}
+
+	st := raven.BuildStackTrace(e.StackTrace)
```

Comment:
crossing dependencies like this isn't great.. would it be straightforward to pull out the BuildStackTrace into another package?  e.g. stacktrace.Build()